server/manager/src/main/java/org/apache/accumulo/manager/tableOps/bulkVer2/LoadFiles.java

```diff
@@ -157,62 +160,149 @@ void start(Path bulkDir, Manager manager, long tid, boolean setTime) throws Exce
+      }
+    }
+
     private void sendQueued(int threshhold) {
       if (queuedDataSize > threshhold || threshhold == 0) {
-        loadQueue.forEach((server, tabletFiles) -> {
+        var sendTimer = Timer.startNew();
+
+        List<Client> clients = new ArrayList<>();
+        try {
+
+          // Send load messages to tablet servers spinning up work, but do not wait on results.
+          loadQueue.forEach((server, tabletFiles) -> {
+
+            if (log.isTraceEnabled()) {
+              log.trace("{} asking {} to bulk import {} files for {} tablets", fmtTid, server,
+                  tabletFiles.values().stream().mapToInt(Map::size).sum(), tabletFiles.size());
+            }
+
+            // On the server side tablets are processed serially with a write to the metadata table
+            // done for each tablet. Chunk the work up for a tablet server up so that it can be sent
+            // over multiple connections allowing it to run in parallel on the server side. This
+            // allows multiple threads on a single tserver to do metadata writes for this bulk
+            // import.
+            int neededConnections = Math.min(maxConnections, tabletFiles.size());
+            List<Map<TKeyExtent,Map<String,MapFileInfo>>> chunks =
+                new ArrayList<>(neededConnections);
+            for (int i = 0; i < neededConnections; i++) {
+              chunks.add(new HashMap<>());
+            }
+
+            int nextConnection = 0;
+            for (var entry : tabletFiles.entrySet()) {
+              chunks.get(nextConnection++ % chunks.size()).put(entry.getKey(), entry.getValue());
+            }
+
+            for (var chunk : chunks) {
+              try {
+                var client = ThriftUtil.getClient(ThriftClientTypes.TABLET_SERVER, server,
+                    manager.getContext(), timeInMillis);
+                // add client to list before calling send in case there is an exception, this makes
+                // sure its returned in the finally
+                clients.add(new Client(server, client));
+                client.send_loadFilesV2(TraceUtil.traceInfo(), manager.getContext().rpcCreds(), tid,
+                    bulkDir.toString(), chunk, setTime);
+              } catch (TException ex) {
+                log.debug("rpc failed server: {}, {}", server, fmtTid, ex);
+              }
+            }
+          });
+
+          long sendTime = sendTimer.elapsed(TimeUnit.MILLISECONDS);
+          sendTimer.restart();
+
+          int outdatedTservers = 0;
+
+          // wait for all the tservers to complete processing
+          for (var client : clients) {
+            try {
+              client.service.recv_loadFilesV2();
+            } catch (TException ex) {
+              String additionalInfo = "";
+              if (ex instanceof TApplicationException) {
+                if (((TApplicationException) ex).getType()
+                    == TApplicationException.UNKNOWN_METHOD) {
```

Comment:
Is this type check and cast able to be done in a single if statement?

```suggestion
              if (ex instanceof TApplicationException && ((TApplicationException) ex).getType()
                   == TApplicationException.UNKNOWN_METHOD) {
```